modApiServer/src/org/aion/api/server/rpc/ApiWeb3Aion.java

```diff
@@ -2648,8 +2660,7 @@ public RpcMsg stratum_getdifficulty() {
     }
 
     public RpcMsg stratum_getmininginfo() {
-        // TODO: [Unity] This cast should be removed when we support staking blocks
-        AionBlock bestBlock = (AionBlock) getBestBlock();
+        Block bestBlock =  getBestBlock();
```

Comment:
I suspect this API should leave for the mining block, not staking block. Therefore, it probably requires to find the previous mining block instead of bestblock